### PR TITLE
Faraz/lp 1199

### PR DESCRIPTION
**Pull request by ahmed-faraz46**

### CODE REVIEW

File: lms/djangoapps/philu_api/helpers.py [hunk 1]
```diff
@@ -44,6 +45,16 @@ def get_social_sharing_urls(course_url, meta_tags):
     }
 
 
+def get_twitter_sharing_url(course_url, meta_tags, tweet_text):
```
Comment (ahmed-faraz46) [inline lms/djangoapps/philu_api/helpers.py]: @muak50 after receiving above mentioned param in `get_social_sharing_urls` we can remove this function.

File: openedx/features/student_certificates/urls.py [hunk 2]
```diff
@@ -7,4 +7,6 @@
 
 urlpatterns = [
     url(r"^certificates/$", views.student_certificates, name="certificates"),
+    url(r"^shared_certificates/(?P<certificate_uuid>[0-9a-f]{32})$", views.shared_student_certificate,
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/urls.py]: @muak50 let's renamed this to `achievements`

File: openedx/features/student_certificates/views.py [hunk 3]
```diff
@@ -93,17 +99,73 @@ def student_certificates(request):
...
+        'user_certificates': user_certificates,
     }
 
     response = render_to_response('certificates.html', context)
     return response
+
+
+@login_required
+@ensure_csrf_cookie
+def shared_student_certificate(request, certificate_uuid):
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/views.py]: @muak50 let's also rename this according to url rename.

File: openedx/features/student_certificates/views.py [hunk 4]
```diff
@@ -2,18 +2,24 @@
 
 from django.conf import settings
 from django.contrib.auth.decorators import login_required
+from django.http import Http404
 from django.views.decorators.csrf import ensure_csrf_cookie
 from django.core.exceptions import ObjectDoesNotExist
 
 from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
 from edxmako.shortcuts import render_to_response
+from lms.djangoapps.philu_api.helpers import get_course_custom_settings, get_social_sharing_urls, get_twitter_sharing_url
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/views.py]: @muak50 let's split it to multiple lines.

File: openedx/features/student_certificates/views.py [hunk 5]
```diff
@@ -93,17 +99,73 @@ def student_certificates(request):
         except TypeError as ex:
             course_title = course.display_name
 
+        custom_settings = get_course_custom_settings(course_id)
+        meta_tags = custom_settings.get_course_meta_tags()
+
+        tweet_text = meta_tags['title'] or "I just completed @PhilanthropyUni's free online %s course and " \
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/views.py]: @muak50 let's define formatter for it.

File: openedx/features/student_certificates/views.py [hunk 6]
```diff
@@ -93,17 +99,73 @@ def student_certificates(request):
...
+                                                   "earned this certificate. Start learning today: %s%s%s%s" % (
+            course.display_name, settings.LMS_ROOT_URL, '/courses/', course.id, '/about')
+
+        meta_tags['title'] = "I just completed Philanthropy University\'s %s %s" % (course.display_name, 'course! ')
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/views.py]: @muak50 let's define formatter for it.

File: openedx/features/student_certificates/views.py [hunk 7]
```diff
@@ -93,17 +99,73 @@ def student_certificates(request):
...
+
+        social_sharing_urls = get_social_sharing_urls(
+            settings.LMS_ROOT_URL + "/shared_certificates/" + certificate.verify_uuid, meta_tags)
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/views.py]: @muak50 let's use `.format`

File: openedx/features/student_certificates/views.py [hunk 8]
```diff
@@ -93,17 +99,73 @@ def student_certificates(request):
...
+
+        social_sharing_urls['twitter'] = get_twitter_sharing_url(
+            settings.LMS_ROOT_URL + "/shared_certificates/" + certificate.verify_uuid, meta_tags, tweet_text)
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/views.py]: @muak50 let's use `.format`

File: openedx/features/student_certificates/views.py [hunk 9]
```diff
@@ -93,17 +99,73 @@ def student_certificates(request):
...
+    except GeneratedCertificate.DoesNotExist:
+        raise Http404
+
+    course = get_course(certificate.course_id)
+
+    custom_settings = get_course_custom_settings(course.id)
+    meta_tags = custom_settings.get_course_meta_tags()
+
+    meta_tags['description'] = meta_tags['description'] or ""
+    meta_tags['title'] = "I just completed Philanthropy University\'s %s %s" % (course.display_name, 'course! ')
```
Comment (ahmed-faraz46) [inline openedx/features/student_certificates/views.py]: @muak50 Let's use same formater here

File: openedx/features/student_certificates/urls.py (hunk 2 above)
Comment (emraan004) [reply openedx/features/student_certificates/urls.py]: > @muak50 let's renamed this to `achievements`

actually keep it singular as `achievement`

Comment (ahmed-faraz46) [reply openedx/features/student_certificates/urls.py]: @emraan004 actually `achievements/uuid` is the complete url, So uuid is something which makes it singular.
We have copied this pattern from edX's url `certificates/uuid`
Please let us know is still need to make it singular.

**Comment by ahmed-faraz46** [comment]
closing the PR, branch needs to be deleted

**Pull request closed without merging**